scripts/prepare-package-for-publish.mjs

```diff
@@ -1,32 +1,111 @@
 import fse from 'fs-extra';
 import path from 'path';
+import glob from 'glob';
+import { displayErrorMessage } from './utils/console.mjs';
 
-const TARGET_PATH = './tmp';
-const filesToMove = [
-  'dist/handsontable.css',
-  'dist/handsontable.full.css',
-  'dist/handsontable.full.js',
-  'dist/handsontable.full.min.css',
-  'dist/handsontable.full.min.js',
-  'dist/handsontable.js',
-  'dist/handsontable.min.css',
-  'dist/handsontable.min.js',
-  'dist/languages',
-  'dist/README.md',
-  'languages',
-  'base.d.ts',
-  'CHANGELOG.md',
-  'handsontable-general-terms.pdf',
-  'handsontable-non-commercial-license.pdf',
-  'handsontable.d.ts',
-  'LICENSE.txt',
-  'package.json',
-  'README.md',
-];
-
-filesToMove.forEach((file) => {
+const TARGET_PATH = './tmp/';
+const PACKAGE_PATH = path.resolve('package.json');
+const DEV_PACKAGE = fse.readJsonSync(PACKAGE_PATH, { encoding: 'utf-8' });
+const { handsontable } = DEV_PACKAGE;
+const {
+  copy: FILES_TO_COPY,
+  exports: EXPORTS_RULES,
+  fields: PACKAGE_FIELDS_TO_COPY,
+} = handsontable;
+
+/**
+ * Copy necessary files we don't need to process.
+ */
+FILES_TO_COPY.forEach((file) => {
   fse.copySync(
     path.resolve(`./${file}`),
-    path.resolve(`${TARGET_PATH}/${file}`),
+    path.resolve(`${TARGET_PATH}${file}`),
     { overwrite: true });
 });
+
+/**
+ * Prepare exports basing on wildcards in paths.
+ */
+const regexpJSFiles = /\.(m|)js$/;
+const groupedExports = EXPORTS_RULES.flatMap((rule) => {
+  if (typeof rule !== 'string') {
+    return rule;
+  }
+
+  const rules = {};
+  const foundFiles = glob.sync(`${rule}`, { cwd: TARGET_PATH });
+
+  foundFiles.forEach((filePath) => {
+    if (!filePath.startsWith('./dist/') && regexpJSFiles.test(filePath)) {
+      const cleanPath = filePath.replace(regexpJSFiles, '').replace('/index', '');
+
+      if (!rules[cleanPath]) {
+        rules[cleanPath] = {};
+      }
+
+      const key = filePath.endsWith('.mjs') ? 'import' : 'require';
+
+      rules[cleanPath][key] = filePath;
+
+    } else {
+      rules[filePath] = filePath;
+    }
+  });
+
+  return rules;
+});
+const targetExports = Object.assign({}, ...groupedExports);
+
+/**
+ * Test exports to verify if paths exist in the target directory.
+ */
+const EXPORTS_ERRORS = [];
+
+Object.keys(targetExports).forEach((ruleName) => {
+  const rule = targetExports[ruleName];
+
+  if (typeof rule === 'string') {
```

Comment:
I wonder does this condition after all these changes (after core review) has reasons to exist? The `rule` is always an object or I missed something?

Reply:
There might be a simple assignment for `./dist/*` files, e.g.:
```
"./i18n/languages/zh-TW": {
  "require": "./i18n/languages/zh-TW.js",
  "import": "./i18n/languages/zh-TW.mjs"
},
"./dist/handsontable.css": "./dist/handsontable.css",
```
what means:
`exports["./i18n/languages/zh-TW"]` is an object with `require` and `import` keys.
`exports["./dist/handsontable.css"]` is a string.

In the future, we might try to optimize usage of `glob` because this package also verifies if files exist under the passed rule.